mlflow/pipelines/utils/execution.py

```diff
@@ -19,6 +19,7 @@ def run_pipeline_step(
     pipeline_root_path: str,
     pipeline_steps: List[BaseStep],
     target_step: BaseStep,
+    template: str = None,
```

Comment:
Add comments to explain what this new `template` is, and how to use it. Also explain what's the behavior when it's `None`.

Reply:
Removing the default value since we throw an exception now if the template is invalid instead of defaulting to the regression template.